src/multisig/keygen/r1.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{
+    collections::TypedUsize,
+    crypto_tools::{k256_serde, rng},
+    multisig,
+    sdk::{
+        api::TofnResult,
+        implementer_api::{serialize, ProtocolBuilder, RoundBuilder},
+    },
+};
+use serde::{Deserialize, Serialize};
+
+use super::{r2, KeygenPartyShareCounts, KeygenProtocolBuilder, KeygenShareId};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub(super) struct Bcast {
+    pub(super) verifying_key: k256_serde::VerifyingKey,
+}
+
+pub(super) fn start(
+    my_keygen_id: TypedUsize<KeygenShareId>,
+    threshold: usize,
+    party_share_counts: KeygenPartyShareCounts,
+    secret_recovery_key: &rng::SecretRecoveryKey,
+    session_nonce: &[u8],
+) -> TofnResult<KeygenProtocolBuilder> {
+    let rng = rng::rng_seed(
+        multisig::KEYGEN_TAG,
```

Comment:
`rng_seed` needs another layer of domain separation to distinguish between multisig, and tss. Either we can use a distinct tag value (i.e `KEYGEN_TAG != 0x00`), or have another `application_tag` param?

Reply:
The main downside with that approach is that the user is even more responsible for making sure they're using the appropriate domain separation. I don't mind switching to it for convenience though since we only do this in a couple of places.

Reply:
Actually, even if we add layers we still run into the same problem.  (Imagine different blocks of code each implementing their own layers that all start with `0x00`.  You could still have a bunch of domains that are all zeros.

If this approach is going to work then you need a central repository of all domains, which negates any benefit of layering.

Perhaps a better approach is to hard-code a randomly-selected 32-byte blob and rely on randomness to ensure global uniqueness.